lib/src/parsers/utilities/parser-utility.ts

```diff
@@ -462,12 +469,43 @@ export function getTargetDeclarationFromTypeReference(
     }
   }
   const targetDeclaration = declarations[0];
+  if (
+    TypeGuards.isInterfaceDeclaration(targetDeclaration) &&
+    targetDeclaration.getIndexSignatures().length > 0
```

Comment:
What is an index signature? And why checking for them here?

Reply:
Added clarifying comments :)